### ISSUE TITLE
Having a isDesktop prop

### ISSUE THREAD

**Issue by hundsim** [ehellman/react-breakpoints#9]
To DRY up the code, I'd like to avoid doing this in every component:

`screenWidth >= breakpoints.desktop`

Any suggestion how/where I could have this comparison predefined?

**Comment by ehellman** [comment]
If you look at my 3.0 PR that I will try to get merged this week (just have to write documentation) I have introduced `snapMode` to avoid re-renders. So it would look like this now:

```js
breakpoints.current >= breakpoints.desktop
```

So if you want that feature you could write your own functions:

```js
// utils.js
export default function isDesktop(breakpoints) {
  return breakpoints.current >= breakpoints.desktop
}

// component.js
import { isDesktop } from 'utils'

const Test = ({ breakpoints }) => (
  <div>Is it desktop? {isDesktop(breakpoints)}</div>
)
```

I think it's important that this is something users build themselves, because not everyone will have a breakpoint named Desktop. 😃

... Just to play on this idea a bit more, in the future we could open up:
```js
<ReactBreakpoints 
  breakpoints={breakpoints}
  helpers={breakpointHelpers}
>
  . . .
```

Then we can add `breakpoints.helpers` to the object passed down via context. It's worth thinking about at least :)

**Comment by hundsim** [comment]
Wow, very promising enhancement indeed!

PS. Och stort tack för välskriven kod, bra intro för mig till `Context` API:et. 😉🙏

**Comment by ehellman** [comment]
Tack för de fina orden! :)

I think your best bet is to use comparison here, instead of creating extra functions. I played with the idea of passing down an object of helper functions but realized that you could do this instead (the example below is written with the 3.0 API):

```js

// these are the breakpoints that you fed ReactBreakpoints
const breakpoints = {
  mobile: 320,
  desktop: 1024
}

// in your component (with renderProps (recommended for most use cases!))
import { Media } from 'react-breakpoints'
class MyComponent extends React.Component {
  render() {
    return (
      <div>
        <Media>
          {({ breakpoints, currentBreakpoint }) => (
            <div>
              {(currentBreakpoint === breakpoints.desktop) && <span>is desktop!</span>
            </div>
          )}
        </Media>
      </div>
    )
  }
}
export default withBreakpoints(MyComponent)

// in your component (with HOC)

import { withBreakpoints } from 'react-breakpoints'
class MyComponent extends React.Component {
  render() {
    const { breakpoints, currentBreakpoint } = this.props
    return (
      <div>
        {(currentBreakpoint === breakpoints.desktop) && <span>is desktop!</span>
      </div>
    )
  }
}
export default withBreakpoints(MyComponent)
```

Because in 3.0 `currentBreakpoint` changes to match one of the keys in your `breakpoints` object, it will be a lot easier to track and you will have to do less comparisons :)

I still have to fix an issue with `refs` before I can merge it, but should be able to merge it during the weekend or on monday!

If you want more examples, you can read the documentation in my pull request [HERE](https://github.com/ehellman/react-breakpoints/blob/e3d6436837a98a4c4473577dc4522b0ffedc0153/README.md)

I have also included an example project in 3.0 that you can look at, available in the PR as well :)

**Comment by ehellman** [comment]
3.0.0 is merged, update! :)
